File: brasil_burger/src/main/java/sn/ism/brasilburger/repositories/interfaces/IMenuRepository.java
package sn.ism.brasilburger.repositories.interfaces;

import sn.ism.brasilburger.entity.Menu;
import sn.ism.brasilburger.entity.CompositionMenu;
import java.util.List;

public interface IMenuRepository extends IRepository<Menu, Integer> {
    boolean addComposition(CompositionMenu composition);
    List<CompositionMenu> findCompositionsByMenuId(Integer menuId);
    double calculatePrixTotal(Integer menuId);
    boolean toggleArchive(Integer id);
}

File: brasilb/Repositories/Interfaces/IMenuRepository.cs
using BrasilBurger.Models;

namespace BrasilBurger.Repositories.Interfaces
{
    public interface IMenuRepository
    {
        List<Menu> GetAll();
        List<Menu> GetNonArchived();
        Menu? GetById(int id);
        List<CompositionMenu> GetCompositions(int idMenu);
        decimal GetPrixTotal(int idMenu);
        
    }
}